chef/cookbooks/ceph/recipes/conf.rb

```diff
@@ -59,22 +59,34 @@
   num_osds = node["ceph"]["config"]["osds_in_total"]
 end
 
-# calculate pg_num based on documentation
-# http://ceph.com/docs/master/rados/operations/placement-groups/
-case num_osds
-when 1..4
-  pg_num = 128
-when 5..9
-  pg_num = 512
-when 10..49
-  pg_num = 4096
-else
-  # Ensure you have a realistic number of placement groups. We recommend
-  # approximately 100 per OSD. E.g., total number of OSDs multiplied by 100
-  # divided by the number of replicas (i.e., osd pool default size).
-  # The result should be rounded up to the nearest power of two.
-  pg_num = 2 ** Math.log2(num_osds * 100 / rep_num).round
-end
+# There'll always be at least the default rbd pool (which Ceph always creates
+# with 64 PGs, irrespective of what "osd pool default pg num" is set to)
+expected_pools = 1
+
+rgw_host = search(:node, "roles:ceph-radosgw")
+# According to http://docs.ceph.com/docs/master/radosgw/config/, radosgw uses
+# 13 pools.  http://ceph.com/pgcalc/ suggests radosgw uses 12 pools.  Random
+# experimentation suggests it might actually be 11 pools (at least on Ceph
+# Jewel).  So let's pretend it's 12.
+expected_pools += 12 unless rgw_host.empty?
```

Comment:
if you select RGW _only_ in pgcalc, you would see 14 pools, 2 of which (.usage, buckets.extra) are not created by default, also most pools are created during run time, for eg the swift pool is created on the fly when you create a swift subuser for the first time, the usage pool is only created if you explicitly configure `rgw enable usage log` in ceph.conf, etc